ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/StateMachine.hs

```diff
@@ -1567,6 +1572,22 @@ prop_trace trace = invalidBlockNeverValidatedAgain
     isOpened (TraceOpenEvent (OpenedDB {})) = True
     isOpened _                              = False
 
+    tentativeHeaderMonotonicity =
+        counterexample "Trap tentative headers did not improve monotonically" $
+        all isMonotonicallyIncreasing trapTentativeSelectViews
+      where
+        isMonotonicallyIncreasing as = and $ zipWith (<) as (drop 1 as)
```

Comment:
Huh. There really should be an `adjacencies :: [a] -> [(a,a)]` function in `base`... maybe we also add that to `Test.Util.Split` and rename the module to `Test.Util.List`?

Reply:
Actually, I just became (again) aware of the already-existing [`strictlyIncreasing`](https://input-output-hk.github.io/ouroboros-network/ouroboros-consensus-test/Test-Util-QuickCheck.html#v:strictlyIncreasing), which I think is appropriate here. Curiously, it is written without `zipWith`, but rather with explicit recursion.

But this discussion is actually not superseded, as I also introduced a [`strictlyIncreasing`](https://github.com/input-output-hk/ouroboros-network/pull/3856/files#diff-171ff84e07dc7c6488cf8adc2cdc77404679523f5e772cd9153072ea711d4c20R248) function (`Bool`-valued) in #3856 (but this time, with `tail` instead of `drop 1` :laughing:). Moving it to some util module sounds good :+1: